packages/lib-classifier/src/store/TutorialStore.spec.js

```diff
@@ -495,73 +495,39 @@ describe('Model > TutorialStore', function () {
+    let tutorialsClient
+
     before(function () {
       const panoptesClientWithUPP = stubPanoptesJs({ project_preferences: upp, workflows: workflow })
       const panoptesClientWithUPPTimestamp = stubPanoptesJs({ project_preferences: uppWithTutorialTimeStamp, workflows: workflow })
-      const tutorialsClient = { 
+      tutorialsClient = { 
         tutorials: {
           get: sinon.stub().callsFake(() => {
             return Promise.resolve({
               body: {
                 tutorials: [tutorial]
               }
             })
-          }),
-          getAttachedImages: sinon.stub().callsFake(() => {
-            return Promise.resolve({
-              body: {
-                media: [medium]
-              }
-            })
           })
         }
       }
       clientStubWithUPP = Object.assign({}, tutorialsClient, panoptesClientWithUPP)
       clientStubWithUPPTimestamp = Object.assign({}, tutorialsClient, panoptesClientWithUPPTimestamp)
     })
-    
-    describe('for logged-out users', function () {
-      it('should show the tutorial', function (done) {
-        rootStore = RootStore.create({
-          projects: ProjectStore.create(),
-          tutorials: TutorialStore.create(),
-          workflows: WorkflowStore.create()
-        }, {
-          authClient: authClientStubWithoutUser, client: clientStub()
-        })
-        const setActiveTutorialSpy = sinon.spy(rootStore.tutorials, 'setActiveTutorial')
-        const setModalVisibilitySpy = sinon.spy(rootStore.tutorials, 'setModalVisibility')
-        sinon.stub(rootStore.userProjectPreferences, 'updateUPP').callsFake(() => {})
-        sinon.stub(rootStore.tutorials, 'fetchTutorials')
 
-        rootStore.projects.setResource(project)
-        rootStore.projects.setActive(project.id)
-        rootStore.workflows.setActive(workflow.id)
-          .then(() => {
-            rootStore.tutorials.fetchTutorials.restore()
-            return rootStore.tutorials.fetchTutorials()
-          })
-          .then(() => {
-            expect(setActiveTutorialSpy).to.have.been.calledOnceWith(tutorial.id)
-            expect(setModalVisibilitySpy).to.have.been.calledOnce
-            expect(rootStore.tutorials.active).to.deep.equal(tutorial)
-            expect(rootStore.tutorials.showModal).to.be.true
-          }).then(() => {
-            setActiveTutorialSpy.restore()
-            setModalVisibilitySpy.restore()
-            rootStore.userProjectPreferences.updateUPP.restore()
-          }).then(done, done)
+    describe('when media have not loaded', function () {
+      before(function () {
+        tutorialsClient.tutorials.getAttachedImages = sinon.stub().callsFake(() => {
+          return Promise.reject('Media could not be loaded')
+        })
       })
-    })
 
-    describe('for logged-in users', function () {
-      it('should show the tutorial if it has not been seen', function (done) {
+      it('should not show the tutorial', function (done) {
```

Comment:
I don't think this should happen. We should allow the tutorial to load if the tutorial does not have any media. This is the current PFE behavior. Do you think that should change?

Reply:
No. My reading of the store code was that the tutorial wasn’t intended to be shown on error. Either way, the code to show a tutorial is crashing if the media map is empty.

Reply:
Hm, I think that's an unintended consequence of the yield for the media requests. I think we should render the tutorial anyway even if the media request errors.

Reply:
I think it’s crashing because the active tutorial step is linked to an image that didn’t load, giving that unresolved reference error.